feat: Add new provider snowflake

tiaz0128 opened the pull request:
@filipeaaoliveira 

Hello. I'm attempting to resolve this issue. #991 

I'm testing to add a new provider 'snowflake'.
I've created a folder 'resources/snowflake/workload' and added PNG files that are under 256px in size.

```text
resources/
  └── snowflake/
          ├── bugworkload
          └── workload
```

Since this is a new provider, I added 'snowflake' to the providers list in the autogen.sh file.
Following the Development Guide, I ran the script autogen.sh.

However, while the image files were successfully copied to website/static/img/resources/snowflake,
it seems that the class files are not being generated in diagrams/snowflake.

Would you please help me understand what might be wrong?

filipeaaoliveira commented:
Hi @tiaz0128 
I did not forget about this, I haven't had the time to look into it, but I'll try to do it this will and will get back to you. 
Sorry for the delay.

tiaz0128 commented:
@filipeaaoliveira @mingrammer 

### Adding New Provider

I believe I've successfully added a new provider. 
There are still a few modifications that might be needed:

1. Adjusting resource icon sizes
2. Adding resource icons
3. Verifying if generated files follow naming conventions

I plan to work on these improvements.
I also want to document the process of adding new providers.
Eventually, I'd like to enhance the scripts to automate provider integration, though this isn't an immediate priority.

### Creating diagrams Folder

I noticed that when adding a provider, a new provider folder isn't automatically created under the diagrams directory, so I added this functionality.

The init file also isn't being generated, so I created it manually for now. If this needs to be done manually anyway, I think we should remove this functionality.

```
os.makedirs(os.path.dirname(mod_path), exist_ok=True)
```

### Bug Fix?
I also discovered and fixed what appears to be a minor bug in the config.py file:

```
"pve": ("pve"),
"ibm": ("ibm"),
```

Would you like me to help you write this as a GitHub issue or pull request description? I can format it specifically for that purpose if needed.

<img width="400px" src="https://github.com/user-attachments/assets/1dee6ce0-df84-4449-8eb0-dc3a9c4a1598" />
<img width="400px" src="https://github.com/user-attachments/assets/5bf84486-4120-492e-a57a-4844a1ca0844" />

gabriel-tessier commented:
Hi @tiaz0128, 

Thank you for the PR,

Check the files changed in the following PR and apply them to yours:
https://github.com/mingrammer/diagrams/pull/974

Can you revert the changes that are not related to the provider changes, it will make it more easy for us to review. If you find bugs open another PR or issue.

Thanks.

tiaz0128 commented:
@gabriel-tessier 

I understand! Thank you for your comments! I will make the changes and try again.

gabriel-tessier reviewed (changes requested):
Left comment, just set the request changes status to prevent merging by mistake.
